plugins/module_utils/config_utils.py

```diff
@@ -99,3 +104,192 @@ def changed(self) -> bool:
+        it will use the version_map attribute based on the instance's version attribute. Defaults to None.
+
+        Returns:
+        - str or None: The XPath as a string if the module and setting are found within the map. If the
+        module or setting cannot be found, or the inputs are None, it returns None.
+
+        Example:
+        - For an instance with version 'OPNsense 23.1', if version_map['OPNsense 23.1'] contains {'system': {'hostname': 'system/hostname'}},
+        and we call _get_xpath(module='system', setting='hostname'), it will return 'system/hostname'.
+        """
+
+        map_dict = self.version_map.get(self.version)
+
+        if map_dict is None:
+            raise KeyError(f"{self.version} was not not found in version_map")
+
+        # Check if the provided module is in the map
+        if module in map_dict:
+            # If the setting is directly within the module, return it
+            if setting in map_dict[module]:
+                return map_dict[module][setting]
+            # If the setting is nested, search recursively
+            else:
+                return self._search_map(map_dict[module], setting)
+
+        return None
+
+    def _get_php_requirements(self, module: str = None, setting: str = None) -> list:
+        """
+        Retrive list of php_requirements for a given module and setting from the version-specific mapping.
+        """
+
+        map_dict = self.version_map.get(self.version)
+
+        if map_dict is None:
+            raise KeyError(f"{self.version} was not not found in version_map")
+
+        # Check if the provided module is in the map
+        if module in map_dict:
+            # If the setting is directly within the module, return it
+            if setting in map_dict[module]:
+                return map_dict[module][setting]
+            # If the setting is nested, search recursively
+            else:
+                return self._search_map(map_dict[module], setting)
+
+        return None
+
+    def _get_configure_functions(self, module: str = None, setting: str = None) -> dict:
+        """
+        Retrive list of configure_functions for a given module and setting from the version-specific mapping.
+        """
+
+        map_dict = self.version_map.get(self.version)
+
+        if map_dict is None:
+            raise KeyError(f"{self.version} was not not found in version_map")
+
+        # Check if the provided module is in the map
+        if module in map_dict and setting in map_dict[module]:
+            # Check if the setting contains a dictionary of configure functions
+            if isinstance(map_dict[module][setting], dict):
+                return map_dict[module][setting]
+
+    def set_module_setting(self, value: str, module: str = None, setting: str = None):
+        """
+        utility to set config specific setting
+        """
+
+        # get xpath from key_mapping
+        xpath = self._get_xpath(module=module, setting=setting)
```

Comment:
`_get_xpath` might return `None` in which case `xpath.split("/")` will result in an `AttributeError` Exception later. Maybe add some error handling here?